relayer/relays/beacon/parachain-writer.go

```diff
@@ -0,0 +1,146 @@
+package beacon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/go-substrate-rpc-client/v4/types"
+	"github.com/snowfork/snowbridge/relayer/chain"
+	"github.com/snowfork/snowbridge/relayer/chain/parachain"
+	"golang.org/x/sync/errgroup"
+)
+
+type InitialSync struct {
+	Header                     interface{}
+	CurrentSyncCommittee       interface{}
+	CurrentSyncCommitteeBranch interface{}
+	ValidatorsRoot             interface{}
+}
+
+type ParachainPayload struct {
+	InitialSync *InitialSync
+	Messages    []*chain.EthereumOutboundMessage
+}
+
+type ParachainWriter struct {
+	conn        *parachain.Connection
+	nonce       uint32
+	pool        *parachain.ExtrinsicPool
+	genesisHash types.Hash
+}
+
+func NewParachainWriter(
+	conn *parachain.Connection,
+) *ParachainWriter {
+	return &ParachainWriter{
+		conn: conn,
+	}
+}
+
+func (wr *ParachainWriter) WritePayload(ctx context.Context, payload *ParachainPayload, eg *errgroup.Group) error {
+	call, err := wr.makeInitialSyncCall(payload.InitialSync)
+	if err != nil {
+		return err
+	}
+
+	wr.pool = parachain.NewExtrinsicPool(eg, wr.conn)
```

Comment:
the pool should only be initialized once, on startup. That said, I think it is more convenient to not use the pool in these early prototyping stages, as it adds complexity.